UnicornEngine/video/include/unicorn/video/geometry/MeshDescriptor.hpp

```diff
@@ -0,0 +1,65 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef UNICORN_VIDEO_GEOMETRY_MESH_DESCRIPTOR_HPP
+#define UNICORN_VIDEO_GEOMETRY_MESH_DESCRIPTOR_HPP
+
+#include <unicorn/utility/SharedMacros.hpp>
+#include <unicorn/video/geometry/Mesh.hpp>
+
+#include <glm/glm.hpp>
+#include <memory>
+
+namespace unicorn
+{
+namespace video
+{
+namespace geometry
+{
+/**
+ * @brief This class is needed for wrapping raw mesh and manipulating it
+ */
+class MeshDescriptor
+{
+public:
+    /**
+     * @brief Mesh Descriptor constructor
+     * @param mesh Pointer to Renderer mesh
+     */
+    UNICORN_EXPORT MeshDescriptor(std::shared_ptr<Mesh> mesh);
```

Comment:
Are you sharing the ownership over mesh or is it controlled from somewhere else? Because if the control is shared, there may be a scenario where a `MeshDescriptor` is operating on the mesh that is no longer processed by the Renderer and therefore irrelevant.